src/amplitude_experiment/deployment/deployment_runner.py

```
@@ -0,0 +1,122 @@
+                continue
+
+            # Keep track of old flag and cohort for each flag
+            old_flag_config = self.flag_config_storage.get_flag_config(flag_config['key'])
+
+            try:
+                flag_loaded = self._load_cohorts_and_store_flag(flag_config, cohort_ids, initial)
+                if flag_loaded:
+                    self.flag_config_storage.put_flag_config(flag_config)  # Store new flag config
+                    self.logger.debug(f"Stored flag config {flag_config['key']}")
+                else:
+                    self.logger.warning(f"Failed to load all cohorts for flag {flag_config['key']}. Using the old flag config.")
+                    self.flag_config_storage.put_flag_config(old_flag_config)
+            except Exception as e:
+                self.logger.error(f"Failed to load cohorts for flag {flag_config['key']}:{e}")
+                if initial:
+                    raise e
+
+        self._delete_unused_cohorts()
+        self.logger.debug(f"Refreshed {len(flag_configs)} flag configs.")
+
+    def _load_cohorts_and_store_flag(self, flag_config: dict, cohort_ids: Set[str], initial: bool):
+        def task():
+            try:
+                for cohort_id in cohort_ids:
+                    future = self.cohort_loader.load_cohort(cohort_id)
+                    future.result()
+                    self.logger.debug(f"Cohort {cohort_id} loaded for flag {flag_config['key']}")
+                return True  # All cohorts loaded successfully
+            except Exception as e:
+                self.logger.error(f"Failed to load cohorts for flag {flag_config['key']}: {e}")
+                if initial:
+                    raise e
+                return False  # Cohort loading failed
+
+        cohort_fetched = self.cohort_loader.executor.submit(task)
+        flag_fetched = True
+
+        # Wait for both flag and cohort loading to complete
+        if initial:
+            flag_fetched = cohort_fetched.result()
```

I'd refactor the function to throw an exception regardless rather than returning a boolean, then handle the exception different in the calling function.